src/vma/sock/sockinfo.h

```diff
@@ -169,6 +169,13 @@ class sockinfo : public socket_fd_api, public pkt_rcvr_sink, public pkt_sndr_sou
 #if _BullseyeCoverage
     #pragma BullseyeCoverage on
 #endif
+
+	enum sockinfo_state {
+		SOCKINFO_OPENED,
+		SOCKINFO_CLOSING,
+		SOCKINFO_CLOSED
+	};
+
```

Comment:
We already have two types of "state" in sockinfo_tcp:
https://github.com/Mellanox/libvma/blob/master/src/vma/sock/sockinfo_tcp.h#L63:L86

Why can't we extend one of them instead of adding new one ?

Reply:
They can not be used as far we need intermediate state as closing. There is conflict between **TCP** state machine values and **sockinfo** states.